### PR TITLE
Add compute step documentation

### PR DESCRIPTION
**Pull request by aymkhalil**
Add `compute` step documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +24,9 @@ The `TransformFunction` reads its configuration as `JSON` from the Function `use
 ```json
 {
   "steps": [
+    {
+      "type": "cast", "schema-type": "STRING"
```
Comment (cbornet) [inline README.md:28]: If you cat to string here, there's no point in dropping fields after

File: README.md [hunk 2]
```diff
@@ -186,10 +196,85 @@ Input: `{field1: {field11: value11, field12: value12}} (AVRO)`
...
+| `DATE`            | a date without a time-zone in the [ISO-8601 format](https://www.iso.org/iso-8601-date-and-time-format.html)          | expression1: "2021-12-03"                                                                              |
+| `TIME`            | a time without a time-zone in the [ISO-8601 format](https://www.iso.org/iso-8601-date-and-time-format.html)          | expression1: "20:15:45"                                                                                |
+| `DATETIME`        | a date-time with an offset from UTC in the [ISO-8601 format](https://www.iso.org/iso-8601-date-and-time-format.html) | expression1: "2022-10-02T01:02:03+02:00", expression2: "2019-10-02T01:02:03Z", expression3: "fn:now()" |
+
+##### Example 1
+
+UserConfig: `{"steps": [{"type": "compute", "fields":[
+                {"name": "key.newKeyField",   "expression" : "5*3", "type": "INT32"},"
+                {"name": "value.valueField",  "expression" : "fn:contact(value.valueField, '_suffix')", "type": "STRING"}]}
```
Comment (cbornet) [inline README.md:233]: s/contact/concat

**Comment by cbornet** [comment]
LGTM

**Review by cbornet — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (aymkhalil) [reply README.md:28]: good point, will restore the old order, I naively wanted to match the the above list.

Comment (aymkhalil) [reply README.md:28]: same applies to flatten and compute, maybe it not necessary to have a single example will possible steps. will remove for now, users can always read the doc on each step.

**Pull request merged**